pkg/apis/batch/types.go

```diff
@@ -244,6 +244,16 @@ type CronJobSpec struct {
 	// JobTemplate is the object that describes the job that will be created when
 	// executing a CronJob.
 	JobTemplate JobTemplateSpec
+
+	// The number of successful finished jobs to retain.
+	// This is a pointer to distinguish between explicit zero and not specified.
+	// +optional
+	SuccessfulJobsHistoryLimit *int32
+
+	// The number of failed finished jobs to retain.
+	// This is a pointer to distinguish between explicit zero and not specified.
+	// +optional
+	FailedJobsHistoryLimit *int32
```

Comment:
It was also mentioned [here](https://github.com/kubernetes/kubernetes/issues/41039#issuecomment-277963383) that it's reasonable to have some defaults in place. @smarterclayton do you think 3 for successful and 1 for failed is a reasonable default?